js_modules/dagit/packages/core/src/ticks/SensorDryRun.tsx

```diff
@@ -0,0 +1,361 @@
+    } else if (data?.setSensorCursor) {
+      const error = data.setSensorCursor;
+      SharedToaster.show({
+        intent: 'danger',
+        message: (
+          <Group direction="row" spacing={8}>
+            <div>Could not set cursor value.</div>
+            <ButtonLink
+              color={Colors.White}
+              underline="always"
+              onClick={() => {
+                showCustomAlert({
+                  title: 'Python Error',
+                  body:
+                    error.__typename === 'PythonError' ? (
+                      <PythonErrorInfo error={error} />
+                    ) : (
+                      'Sensor not found'
+                    ),
+                });
+              }}
+            >
+              View error
+            </ButtonLink>
+          </Group>
+        ),
+      });
+    }
+  }, [sensorExecutionData?.evaluationResult?.cursor, sensorSelector, setCursorMutation]);
+
+  const content = React.useMemo(() => {
+    if (sensorExecutionData || error) {
+      const runRequests = sensorExecutionData?.evaluationResult?.runRequests;
+      const numRunRequests = runRequests?.length || 0;
+      const didSkip = !error && numRunRequests === 0;
+      return (
+        <Box flex={{direction: 'column', gap: 8}}>
+          <Box>
+            <Grid>
+              <div>
+                <Subheading>Result</Subheading>
+                <Box flex={{grow: 1, alignItems: 'center'}}>
+                  <div>
+                    {error ? (
+                      <Tag intent="danger">Failed</Tag>
+                    ) : numRunRequests ? (
+                      <Tag intent="success">{numRunRequests} Run requests</Tag>
+                    ) : (
+                      <Tag intent="warning">Skipped</Tag>
+                    )}
+                  </div>
+                </Box>
+              </div>
+              <div>
+                <Subheading>Used cursor value</Subheading>
+                <pre>{currentCursor?.length ? currentCursor : 'None'}</pre>
+              </div>
+              <div>
+                <Subheading>Computed cursor value</Subheading>
+                <pre>
+                  {sensorExecutionData?.evaluationResult?.cursor?.length
+                    ? sensorExecutionData?.evaluationResult.cursor
+                    : error
+                    ? 'Error'
+                    : 'None'}
+                </pre>
+                {error ||
+                (currentCursor ?? '') ===
+                  (sensorExecutionData?.evaluationResult?.cursor ?? 'testing1234') ? null : (
+                  <Box flex={{direction: 'row', gap: 8, alignItems: 'center'}}>
+                    <Button
+                      disabled={['Persisting', 'Persisted'].includes(cursorState)}
+                      loading={cursorState === 'Persisting'}
+                      onClick={onPersistCursorValue}
+                    >
+                      <span data-testid={testId('persist-cursor')}>
```

Comment:
Why test ID on the span instead of the button itself? Just curious

Reply:
the button doesn't propagate the data attribute :(